### ISSUE TITLE
Steam runs only first time

### ISSUE THREAD

**Issue by keleset** [flathub/com.valvesoftware.Steam#602]
Flatpak version: 1.8.1
Distro: Alpine Edge
So, the problem is: Steam installs and runs successfully, but only once. Any subsequent run ends instantly with:
```
 flatpak run -v com.valvesoftware.Steam
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/me/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Cleaning up unused container id 2982173205
F: Allocated instance id 823989937
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/me/.config/pulse/client.conf': Error opening file /home/me/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 33 xdg-dbus-proxy --args=35'
F: Running 'bwrap --args 33 steam-wrapper'
bwrap: Can't bind mount /bindfileBHIoNa on /newroot/home/me/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs: Invalid argument
```
The only thing that helps (to execute it one more time) is `flatpak uninstall --delete-data com.valvesoftware.Steam` and reinstallation.

**Comment by nanonyme** [comment]
Looks like https://github.com/flatpak/flatpak/issues/3807; if you can run flatpak run -vv com.valvesoftware.Steam and attach output to the other issue, that might help

**Comment by keleset** [comment]
Actually mount points in the issues are different, so I'm not sure that this is the same issue. Here:
```
https://pastebin.com/1GMnDbUz
```
I have:
```
bwrap: Can't bind mount /bindfileaCmklc on /newroot/home/me/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs: Invalid argument
```
The other issue is about xdg directories, but mine is about binding `user-dirs.dirs` and `/bindfileaCmklc` (not sure what is `bindfileaCmklc` here, actually this `bindfile` has its suffix generated differently every time).

Maybe I should open another issue in `/flatpak` ?

**Comment by nanonyme** [comment]
Hmm. I wonder if there's something weird happening here because of Steam app making config into a symlink to .config.

**Comment by nanonyme** [comment]
@alexlarsson any idea? This hasn't been happening with old flatpak versions.

**Comment by PureTryOut** [comment]
I'm the author of #3807, but after a re-installation of my PC I have this issue instead. Same file failing, `user-dirs.dirs`, and Steam only starts the first time. This gets annoying quickly...

Strangely enough it works perfectly fine on my laptop running the same OS (Alpine Linux) and it did the same before a reinstall on my desktop.

**Comment by nanonyme** [comment]
Hmm, this user-dirs.dirs seems to be in config dir which gets relocated to .config as part of Steam Cloud fixup.

**Comment by sirn** [comment]
Have been having this issue on Void Linux (musl) as well. I my case this issue was always triggered when I have `xdg-desktop` in overrides (`flatpak override --user --filesystem=xdg-desktop`, including `xdg-desktop:ro`). Removing `xdg-desktop` from overrides fixed the issue for me.

This is what I found so far. Each heading are run with `~/.var/app/com.valvesoftware.Steam` deleted prior to testing. It seems to me that if either Steam or Flatpak tries to create `user-dirs.dirs` if `xdg-desktop` is present.

-------

#### Scenario: no override

First run:

```
$ flatpak run -v com.valvesoftware.Steam
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Regenerating ld.so.cache /home/sirn/.var/app/com.valvesoftware.Steam/.ld.so/f9df2981b5833628e478c8db9eb911b45289a89244ec919900d82f65599ba74f
F: Running: 'bwrap --args 16 ldconfig -X -C /run/ld-so-cache-dir/f9df2981b5833628e478c8db9eb911b45289a89244ec919900d82f65599ba74f'
F: Cleaning up unused container id 4117146237
F: Allocated instance id 3081096011
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/sirn/.config/pulse/client.conf': Error opening file /home/sirn/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 33 /usr/libexec/flatpak-dbus-proxy --args=35'
F: Running 'bwrap --args 33 steam-wrapper'
https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions
Overriding TZ to Asia/Tokyo
Setting up Steam content in /home/sirn/.local/share/Steam
Running Steam on org.freedesktop.platform 20.08.0 64-bit
<snip>

$ ls -alh ~/.var/app/com.valvesoftware.Steam
total 95K
drwxr-xr-x  9 sirn sirn 14 Oct 10 01:21 .
drwxr-xr-x 30 sirn sirn 30 Oct 10 01:20 ..
drwxr-xr-x  5 sirn sirn  6 Oct 10 01:22 .cache
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:21 .config
drwxr-xr-x  2 sirn sirn  4 Oct 10 01:20 .ld.so
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:20 .local
drwxr-xr-x  2 sirn sirn 13 Oct 10 01:22 .steam
lrwxrwxrwx  1 sirn sirn 29 Oct 10 01:21 .steampath -> /home/sirn/.steam/sdk32/steam
lrwxrwxrwx  1 sirn sirn 27 Oct 10 01:21 .steampid -> /home/sirn/.steam/steam.pid
drwxr-xr-x  3 sirn sirn  3 Oct 10 01:20 .var
lrwxrwxrwx  1 sirn sirn  6 Oct 10 01:20 cache -> .cache
lrwxrwxrwx  1 sirn sirn  7 Oct 10 01:20 config -> .config
drwxr-xr-x  2 sirn sirn  2 Oct 10 01:20 config.old
lrwxrwxrwx  1 sirn sirn 12 Oct 10 01:20 data -> .local/share

$ ls -alh ~/.var/app/com.valvesoftware.Steam/config/
total 34K
drwxr-xr-x 4 sirn sirn  4 Oct 10 01:21 .
drwxr-xr-x 9 sirn sirn 14 Oct 10 01:21 ..
drwx------ 3 sirn sirn  3 Oct 10 01:21 cef_user_data
drwx------ 2 sirn sirn  3 Oct 10 01:21 pulse
```

Subsequent runs:

```
$ flatpak run -v com.valvesoftware.Steam
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Cleaning up unused container id 3081096011
F: Allocated instance id 1939282817
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/sirn/.config/pulse/client.conf': Error opening file /home/sirn/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 33 /usr/libexec/flatpak-dbus-proxy --args=35'
F: Running 'bwrap --args 33 steam-wrapper'
https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions
Overriding TZ to Asia/Tokyo
Running Steam on org.freedesktop.platform 20.08.0 64-bit
<snip>

$ ls -alh ~/.var/app/com.valvesoftware.Steam                                                          
total 87K
drwxr-xr-x  8 sirn sirn 13 Oct 10 01:22 .
drwxr-xr-x 30 sirn sirn 30 Oct 10 01:20 ..
drwxr-xr-x  5 sirn sirn  6 Oct 10 01:22 .cache
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:21 .config
drwxr-xr-x  2 sirn sirn  4 Oct 10 01:20 .ld.so
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:20 .local
drwxr-xr-x  2 sirn sirn 13 Oct 10 01:22 .steam
lrwxrwxrwx  1 sirn sirn 29 Oct 10 01:22 .steampath -> /home/sirn/.steam/sdk32/steam
lrwxrwxrwx  1 sirn sirn 27 Oct 10 01:22 .steampid -> /home/sirn/.steam/steam.pid
drwxr-xr-x  3 sirn sirn  3 Oct 10 01:20 .var
lrwxrwxrwx  1 sirn sirn  6 Oct 10 01:20 cache -> .cache
lrwxrwxrwx  1 sirn sirn  7 Oct 10 01:20 config -> .config
lrwxrwxrwx  1 sirn sirn 12 Oct 10 01:20 data -> .local/share

$ ls -alh ~/.var/app/com.valvesoftware.Steam/config/                                                  
total 34K
drwxr-xr-x 4 sirn sirn  4 Oct 10 01:21 .
drwxr-xr-x 8 sirn sirn 13 Oct 10 01:22 ..
drwx------ 3 sirn sirn  3 Oct 10 01:21 cef_user_data
drwx------ 2 sirn sirn  3 Oct 10 01:21 pulse
```

-------

#### Scenario: with `--filesystem=xdg-desktop` override

First run:

```
$ flatpak run -v com.valvesoftware.Steam                                                          
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Regenerating ld.so.cache /home/sirn/.var/app/com.valvesoftware.Steam/.ld.so/f9df2981b5833628e478c8db9eb911b45289a89244ec919900d82f65599ba74f
F: Running: 'bwrap --args 16 ldconfig -X -C /run/ld-so-cache-dir/f9df2981b5833628e478c8db9eb911b45289a89244ec919900d82f65599ba74f'
F: Cleaning up unused container id 1939282817
F: Allocated instance id 3091821156
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/sirn/.config/pulse/client.conf': Error opening file /home/sirn/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 34 /usr/libexec/flatpak-dbus-proxy --args=36'
F: Running 'bwrap --args 34 steam-wrapper'
https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions
Relocating /home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs to .config/user-dirs.dirs
Overriding TZ to Asia/Tokyo
Setting up Steam content in /home/sirn/.local/share/Steam
Running Steam on org.freedesktop.platform 20.08.0 64-bit
<snip>

$ ls -alh ~/.var/app/com.valvesoftware.Steam                                                          
total 104K
drwxr-xr-x 10 sirn sirn 15 Oct 10 01:25 .
drwxr-xr-x 30 sirn sirn 30 Oct 10 01:24 ..
drwxr-xr-x  5 sirn sirn  6 Oct 10 01:25 .cache
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .config
drwxr-xr-x  2 sirn sirn  4 Oct 10 01:24 .ld.so
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:24 .local
drwxr-xr-x  2 sirn sirn 13 Oct 10 01:26 .steam
lrwxrwxrwx  1 sirn sirn 29 Oct 10 01:25 .steampath -> /home/sirn/.steam/sdk32/steam
lrwxrwxrwx  1 sirn sirn 27 Oct 10 01:25 .steampid -> /home/sirn/.steam/steam.pid
drwxr-xr-x  3 sirn sirn  3 Oct 10 01:24 .var
drwxr-xr-x  2 sirn sirn  2 Oct 10 01:24 Desktop
lrwxrwxrwx  1 sirn sirn  6 Oct 10 01:24 cache -> .cache
lrwxrwxrwx  1 sirn sirn  7 Oct 10 01:24 config -> .config
drwxr-xr-x  2 sirn sirn  3 Oct 10 01:24 config.old
lrwxrwxrwx  1 sirn sirn 12 Oct 10 01:24 data -> .local/share

$ ls -alh ~/.var/app/com.valvesoftware.Steam/config/                                                  
total 39K
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .
drwxr-xr-x 10 sirn sirn 15 Oct 10 01:25 ..
drwx------  3 sirn sirn  3 Oct 10 01:25 cef_user_data
drwx------  2 sirn sirn  3 Oct 10 01:25 pulse
-rw-------  1 sirn sirn 37 Oct 10 01:24 user-dirs.dirs

$ cat ~/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs                                        
XDG_DESKTOP_DIR="/home/sirn/Desktop"
```

Subsequent runs:

```
$ flatpak run -v com.valvesoftware.Steam                                                          
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Cleaning up unused container id 3091821156
F: Allocated instance id 2335163646
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/sirn/.config/pulse/client.conf': Error opening file /home/sirn/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 34 /usr/libexec/flatpak-dbus-proxy --args=36'
F: Running 'bwrap --args 34 steam-wrapper'
bwrap: Can't bind mount /bindfileoIPGhl on /newroot/home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs: Invalid argument                                                          

$ ls -alh ~/.var/app/com.valvesoftware.Steam                                                        
total 104K
drwxr-xr-x 10 sirn sirn 15 Oct 10 01:25 .
drwxr-xr-x 30 sirn sirn 30 Oct 10 01:24 ..
drwxr-xr-x  5 sirn sirn  6 Oct 10 01:25 .cache
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .config
drwxr-xr-x  2 sirn sirn  4 Oct 10 01:24 .ld.so
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:24 .local
drwxr-xr-x  2 sirn sirn 13 Oct 10 01:26 .steam
lrwxrwxrwx  1 sirn sirn 29 Oct 10 01:25 .steampath -> /home/sirn/.steam/sdk32/steam
lrwxrwxrwx  1 sirn sirn 27 Oct 10 01:25 .steampid -> /home/sirn/.steam/steam.pid
drwxr-xr-x  3 sirn sirn  3 Oct 10 01:24 .var
drwxr-xr-x  2 sirn sirn  2 Oct 10 01:24 Desktop
lrwxrwxrwx  1 sirn sirn  6 Oct 10 01:24 cache -> .cache
lrwxrwxrwx  1 sirn sirn  7 Oct 10 01:24 config -> .config
drwxr-xr-x  2 sirn sirn  3 Oct 10 01:24 config.old
lrwxrwxrwx  1 sirn sirn 12 Oct 10 01:24 data -> .local/share                                                          

$ ls -alh ~/.var/app/com.valvesoftware.Steam/config/ 
total 39K
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .
drwxr-xr-x 10 sirn sirn 15 Oct 10 01:25 ..
drwx------  3 sirn sirn  3 Oct 10 01:25 cef_user_data
drwx------  2 sirn sirn  3 Oct 10 01:25 pulse
-rw-------  1 sirn sirn 37 Oct 10 01:24 user-dirs.dirs

$ cat ~/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs                                    
XDG_DESKTOP_DIR="/home/sirn/Desktop"

$ ls -alh /home/sirn/.var/app/com.valvesoftware.Steam/config.old/                                     
total 18K
drwxr-xr-x  2 sirn sirn  3 Oct 10 01:24 .
drwxr-xr-x 10 sirn sirn 15 Oct 10 01:25 ..
-rw-rw-rw-  1 sirn sirn  0 Oct 10 01:24 user-dirs.dirs
```

Deleting `config` and `config.old` and re-run:

```
$ rm -rf /home/sirn/.var/app/com.valvesoftware.Steam/config*
$ flatpak run -v com.valvesoftware.Steam 
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/sirn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Cleaning up unused container id 1497066771
F: Allocated instance id 1373407960
F: Add defaults in dir /com/valvesoftware/Steam/
F: Add locks in dir /com/valvesoftware/Steam/
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/sirn/.config/pulse/client.conf': Error opening file /home/sirn/.config/pulse/client.conf: No such file or directory
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running 'bwrap --args 34 /usr/libexec/flatpak-dbus-proxy --args=36'
F: Running 'bwrap --args 34 steam-wrapper'
https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions
Relocating .config/user-dirs.dirs to .config.bak/user-dirs.dirs
Relocating .config/pulse/cookie to .config.bak/pulse/cookie
Relocating /home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs to .config/user-dirs.dirs
Overriding TZ to Asia/Tokyo
Running Steam on org.freedesktop.platform 20.08.0 64-bit
<snip>

$ ls -alh ~/.var/app/com.valvesoftware.Steam                                                          
total 112K
drwxr-xr-x 11 sirn sirn 16 Oct 10 01:29 .
drwxr-xr-x 30 sirn sirn 32 Oct 10 01:27 ..
drwxr-xr-x  5 sirn sirn  7 Oct 10 01:29 .cache
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .config
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:29 .config.bak
drwxr-xr-x  2 sirn sirn  4 Oct 10 01:24 .ld.so
drwxr-xr-x  4 sirn sirn  4 Oct 10 01:24 .local
drwxr-xr-x  2 sirn sirn 13 Oct 10 01:29 .steam
lrwxrwxrwx  1 sirn sirn 29 Oct 10 01:29 .steampath -> /home/sirn/.steam/sdk32/steam
lrwxrwxrwx  1 sirn sirn 27 Oct 10 01:29 .steampid -> /home/sirn/.steam/steam.pid
drwxr-xr-x  3 sirn sirn  3 Oct 10 01:24 .var
drwxr-xr-x  2 sirn sirn  2 Oct 10 01:24 Desktop
lrwxrwxrwx  1 sirn sirn  6 Oct 10 01:24 cache -> .cache
lrwxrwxrwx  1 sirn sirn  7 Oct 10 01:29 config -> .config
drwxr-xr-x  2 sirn sirn  3 Oct 10 01:29 config.old
lrwxrwxrwx  1 sirn sirn 12 Oct 10 01:24 data -> .local/share

$ ls -alh ~/.var/app/com.valvesoftware.Steam/config/                                                  
total 39K
drwxr-xr-x  4 sirn sirn  5 Oct 10 01:25 .
drwxr-xr-x 11 sirn sirn 16 Oct 10 01:29 ..
drwx------  3 sirn sirn  3 Oct 10 01:25 cef_user_data
drwx------  2 sirn sirn  3 Oct 10 01:25 pulse
-rw-------  1 sirn sirn 37 Oct 10 01:29 user-dirs.dirs

$ ls -alh /home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs                           
-rw------- 1 sirn sirn 37 Oct 10 01:29 /home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs

$ cat /home/sirn/.var/app/com.valvesoftware.Steam/config/user-dirs.dirs                               
XDG_DESKTOP_DIR="/home/sirn/Desktop"

$ ls -alh /home/sirn/.var/app/com.valvesoftware.Steam/config.old             
total 18K
drwxr-xr-x  2 sirn sirn  3 Oct 10 01:29 .
drwxr-xr-x 11 sirn sirn 16 Oct 10 01:29 ..
-rw-rw-rw-  1 sirn sirn  0 Oct 10 01:29 user-dirs.dirs
```

**Comment by nanonyme** [comment]
Please do not post such lengthy outputs here. This is an issue in how flatpak/bubblewrap handle bind mounts over symlinks.

**Comment by nanonyme** [comment]
Could someone please test if this still happens with latest stable?

**Comment by PureTryOut** [comment]
I haven't had this issue for quite a while luckily.
